app.js

```diff
@@ -63,27 +65,37 @@ app.use("/api/manauth", loginRoute);
 app.use("/api/preview", previewRoute);
 
 let child;
-var io = socket(server, { path: "/api/socket.io" });
+
+var io = socket(server, {path: '/api/socket.io'});
+
 io.origins("*:*");
 io.on("connection", (person) => {
 	console.log(`made socket connection : ${person.id}`);
 
 	person.on("cmd", function (val) {
+
 		let val_json = JSON.parse(val);
 		let token = val_json.token;
-		let user_id = val_json.user_id;
 		let fileName = val_json.fileId;
 		let data = val_json.data;
-		let email;
 
 		try {
-			email = jwt.verify(token, process.env.JWT_KEY);
+				jwt.verify(token, process.env.JWT_KEY, (err, user) => {
+	      if (err) {
+	        return res.sendStatus(403);
+	      }
+	      let user_id = user[0]._id;
+	      let email = user[0].email;
```

Comment:
Don't send user_id in the token. 
Send only email and check-in DB for user_id. 
He can still execute bash injection in this.